includes/class-wc-adapter.php

```diff
@@ -0,0 +1,682 @@
+			$tax_percent       = ( $tax > 0 ) ? round( 100 / ( $discount / $tax ) ) : 0;
+
+			$items[] = array(
+				OrderItemInterface::FIELD_REFERENCE   => 'discount',
+				OrderItemInterface::FIELD_NAME        => __( 'Discount', 'swedbank-pay-woocommerce-payments' ),
+				OrderItemInterface::FIELD_TYPE        => OrderItemInterface::TYPE_DISCOUNT,
+				OrderItemInterface::FIELD_CLASS       => 'ProductGroup1',
+				OrderItemInterface::FIELD_QTY         => 1,
+				OrderItemInterface::FIELD_QTY_UNIT    => 'pcs',
+				OrderItemInterface::FIELD_UNITPRICE   => round( - 100 * $discount_with_tax ),
+				OrderItemInterface::FIELD_VAT_PERCENT => round( 100 * $tax_percent ),
+				OrderItemInterface::FIELD_AMOUNT      => round( - 100 * $discount_with_tax ),
+				OrderItemInterface::FIELD_VAT_AMOUNT  => round( - 100 * $tax ),
+			);
+		}
+
+		// Payer reference
+		// Get Customer UUID
+		$user_id = $order->get_customer_id();
+		if ( $user_id > 0 ) {
+			$payer_reference = get_user_meta( $user_id, '_payex_customer_uuid', true );
+			if ( empty( $payer_reference ) ) {
+				$payer_reference = $this->get_uuid( $user_id );
+				update_user_meta( $user_id, '_payex_customer_uuid', $payer_reference );
+			}
+		} else {
+			$payer_reference = $this->get_uuid( uniqid( $order->get_billing_email() ) );
+		}
+
+		$shipping_country = isset( $countries[ $order->get_shipping_country() ] ) ? $countries[ $order->get_shipping_country() ] : '';
+		$billing_country  = isset( $countries[ $order->get_billing_country() ] ) ? $countries[ $order->get_billing_country() ] : '';
+
+		return array(
+			OrderInterface::ORDER_ID              => $order->get_id(),
+			OrderInterface::AMOUNT                => apply_filters(
+				'swedbank_pay_order_amount',
+				$order->get_total(),
+				$order
+			),
+			OrderInterface::VAT_AMOUNT            => apply_filters(
+				'swedbank_pay_order_vat',
+				$info['vat_amount'],
+				$order
+			),
+			OrderInterface::VAT_RATE              => 0, // Can be different
+			OrderInterface::SHIPPING_AMOUNT       => 0, // @todo
+			OrderInterface::SHIPPING_VAT_AMOUNT   => 0, // @todo
+			OrderInterface::DESCRIPTION           => sprintf(
+			/* translators: 1: order id */                __( 'Order #%1$s', 'swedbank-pay-woocommerce-payments' ),
+				$order->get_order_number()
+			),
+			OrderInterface::CURRENCY              => $order->get_currency(),
+			OrderInterface::STATUS                => $order->get_status(),
+			OrderInterface::CREATED_AT            => gmdate( 'Y-m-d H:i:s', $order->get_date_created()->getTimestamp() ),
+			OrderInterface::PAYMENT_ID            => $order->get_meta( '_payex_payment_id' ),
+			OrderInterface::PAYMENT_ORDER_ID      => $order->get_meta( '_payex_paymentorder_id' ),
+			OrderInterface::NEEDS_SAVE_TOKEN_FLAG => '1' === $order->get_meta( '_payex_generate_token' ) &&
+													 0 === count( $order->get_payment_tokens() ),
+
+			OrderInterface::HTTP_ACCEPT           => isset( $_SERVER['HTTP_ACCEPT'] ) ? $_SERVER['HTTP_ACCEPT'] : null,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Direct use of $_SERVER Superglobal detected.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-woocommerce-payments/pullRequest?prid=5340032)